src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+using Internal.IO;
+
+namespace System
+{
+    internal static class StartupHookProvider
+    {
+        private const string startupHookTypeName = "StartupHook";
```

Comment:
Our coding style for the constant names is PascalCasing.

https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md